packages/types/src/lib/interfaces.ts

```diff
@@ -214,83 +214,44 @@ export interface ClaimKeyResponse {
 /**
  * Struct in rust
  * -----
- pub struct JsonExecutionRequest {
-    pub code: Option<String>,
-    pub ipfs_id: Option<String>,
-    pub auth_sig: AuthSigItem,
-    pub js_params: Option<serde_json::Value>,
+pub struct JsonExecutionRequest {
+  pub auth_sig: AuthSigItem,
+  #[serde(default = "default_epoch")]
+  pub epoch: u64,
+  
+  pub ipfs_id: Option<String>,
+  pub code: Option<String>,
+    pub js_params: Option<Value>,
+    pub auth_methods: Option<Vec<AuthMethod>>,
 }
  */
-export interface BaseJsonExecutionRequest {
-  // // the authSig to use to authorize the user with the nodes
-  // authSig?: AuthSig;
-
-  // An object that contains params to expose to the Lit Action.  These will be injected to the JS runtime before your code runs, so you can use any of these as normal variables in your Lit Action.
-  jsParams?: any;
-
-  // JS code to run on the nodes
-  code?: string;
-
-  // The IPFS ID of some JS code to run on the nodes
-  ipfsId?: string;
-
-  // // the session signatures to use to authorize the user with the nodes
-  // sessionSigs?: any;
-
-  // whether to run this on a single node or many
-  targetNodeRange?: number;
-
-  // auth methods to resolve
-  authMethods?: AuthMethod[];
-}
-
-export interface WithAuthSig extends BaseJsonExecutionRequest {
-  authSig: AuthSig;
-  sessionSigs?: any;
-}
-
-export interface WithSessionSigs extends BaseJsonExecutionRequest {
-  sessionSigs: any;
-  authSig?: AuthSig;
-}
-
-export type JsonExecutionRequest = WithAuthSig | WithSessionSigs;
-
-export interface JsExecutionRequestBody {
-  authSig?: AuthSig;
-  code?: string;
-  ipfsId?: string;
-  authMethods?: AuthMethod[];
-  jsParams?: any;
-}
 
 export interface BaseJsonPkpSignRequest {
+  authMethods?: AuthMethod[];
   toSign: ArrayLike<number>;
-  pubKey: string;
 }
 
-export interface WithAuthMethodSigning extends BaseJsonPkpSignRequest {
-  // auth methods to resolve
-  authMethods: AuthMethod[];
-  sessionSigs?: any;
-  authSig?: AuthSig;
-}
-export interface WithSessionSigsSigning extends BaseJsonPkpSignRequest {
-  sessionSigs: any;
-  authSig?: AuthSig;
+/**
+ * The 'pkpSign' function param. Please note that the structure
+ * is different than the payload sent to the node.
+ */
+export interface JsonPkpSignSdkParams extends BaseJsonPkpSignRequest {
+  pubKey: string;
+  sessionSigs: SessionSigsMap;
   authMethods?: AuthMethod[];
 }
 
-export interface WithAuthSigSigning extends BaseJsonPkpSignRequest {
+/**
+ * The actual payload structure sent to the node /pkp/sign endpoint.
+ */
+export interface JsonPkpSignRequest extends BaseJsonPkpSignRequest {
```

Comment:
Why two different functions?

Reply:
`fooSdkParams` is the params that we are accepting in the SDK, then within the function we are converting the params into a shape that the node actually accepts, which is `JsonFooRequest`. eg. node doesn't actually accept `sessionSigs` as a key, but it's expecting an `authSig` which we form within the SDK, BUT it's different than the "client-side` authSig which the user generate themselves. cc @joshLong145

Reply:
This makes sense to me since the sdk abstracts the the node api which only has a concept of `authSigs` and sends the relevant signature which is mapped to a node's `url` so we have to type to separate interfaces, one that we expose from the sdk and one that is for our I/O with the nodes.